hassio/src/components/supervisor-snapshot-content.ts

```diff
@@ -0,0 +1,384 @@
+            </div>
+            <ha-formfield
+              .label=${this.supervisor.localize("snapshot.full_snapshot")}
+            >
+              <ha-radio
+                @change=${this._handleRadioValueChanged}
+                value="full"
+                name="snapshotType"
+                .checked=${this.snapshotType === "full"}
+              >
+              </ha-radio>
+            </ha-formfield>
+            <ha-formfield
+              .label=${this.supervisor!.localize("snapshot.partial_snapshot")}
+            >
+              <ha-radio
+                @change=${this._handleRadioValueChanged}
+                value="partial"
+                name="snapshotType"
+                .checked=${this.snapshotType === "partial"}
+              >
+              </ha-radio>
+            </ha-formfield>
+          </div>`
+        : ""}
+      ${this.snapshot && this.snapshotType === "partial"
+        ? html`
+            ${this.snapshot.homeassistant
+              ? html`<div
+                  class="checkbox-row"
+                  @click=${() => {
+                    this.homeAssistant = !this.homeAssistant;
+                  }}
+                >
+                  <ha-checkbox .checked=${this.homeAssistant}> </ha-checkbox>
+                  <ha-svg-icon .path=${mdiHomeAssistant} class="icon">
+                  </ha-svg-icon>
+                  <span class="item-name">Home Assistant</span>
+                  <span class="item-version">
+                    (${this.snapshot.homeassistant})
+                  </span>
+                </div>`
+              : ""}
+          `
+        : ""}
+      ${this.snapshotType === "partial"
+        ? html`
+            ${this.folders?.length
+              ? html`
+                  <div
+                    class="checkbox-row"
+                    @click=${() =>
+                      this._toggleSection("folders", this.folders!)}
+                  >
+                    <ha-checkbox
+                      ?indeterminate=${this._sectionIndeterminate(this.folders)}
+                      .checked=${this._sectionCheked(this.folders)}
+                    >
+                    </ha-checkbox>
+                    <ha-svg-icon .path=${mdiFolder} class="icon"></ha-svg-icon>
+                    ${this.supervisor.localize("snapshot.folders")}
+                  </div>
+                  <div class="section-content">
+                    ${this.folders.map(
+                      (item) => html`
+                        <div
+                          class="checkbox-row"
+                          @click=${() => this._updateFolder(item)}
+                        >
+                          <ha-checkbox .checked=${item.checked}> </ha-checkbox>
+                          <ha-svg-icon .path=${mdiFolder} class="icon">
+                          </ha-svg-icon>
+                          ${item.name}
+                        </div>
+                      `
+                    )}
```

Comment:
If you do this part in a separate function, you can determine `indeterminate` and `checked` without the 2 extra loops.

```ts
private _getFolder() {
	const templates = [];
	let noChecked = 0;
	let checked = false;
    let indeterminate = false
	this.folders.forEach(item => {template.push(html`...`); if (item.checked) noChecked++});
	
    if (noChecked === this.folder.length) checked = true
	else if (noChecked) indeterminate = true
    return {templates, checked, indeterminate}
}
```